### PR TITLE
Add new `Lint/DuplicateMatchPattern` cop

### PR DESCRIPTION
**Pull request by koic**
This PR adds new `Lint/DuplicateMatchPattern` cop. It checks that there are no repeated patterns used in `in` keywords.

```ruby
# bad
case x
in 'first'
  do_something
in 'first'
  do_something_else
end

# good
case x
in 'first'
  do_something
in 'second'
  do_something_else
end
```

I was wondering whether to extend existing `Lint/DuplicateCaseCondition`, but it is a different inspection target from the concept named "`case` condition". So, I implemented it as a new `Lint/DuplicateMatchPattern` for "match pattern".

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/lint/duplicate_in_pattern.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks that there are no repeated patterns used in `in` keywords.
+      #
+      # @example
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/duplicate_in_pattern.rb:8]: It'd be nice to show examples with more complex patterns, so it's clear that this cop is pretty smart about them.

**Comment by bbatsov** [comment]
I wonder if `DuplicateMatchPattern` might not be a better name for the cop (or at least a more descriptive name).

**Comment by koic** [comment]
Thank you for the suggestions. `DuplicateMatchPattern` looks like a better name and I've also updated the examples.

**Comment by bbatsov** [comment]
Looks great. Thanks!

**Pull request merged**